### PR TITLE
Bolts creation script.

### PR DESCRIPTION
**Pull request by Dartellum**
<details>
  <summary>Click to view bolt creation log.</summary>

```
2021-02-14 17:11:50 -0500:>;bolts cougar-claw maple
2021-02-14 17:11:50 -0500:--- Lich: bolts active.
2021-02-14 17:11:50 -0500:[bolts]>tap my maple lumber
2021-02-14 17:11:50 -0500:I could not find what you were referring to.
2021-02-14 17:11:50 -0500:[bolts]>tap my shafts
2021-02-14 17:11:50 -0500:You tap some bolt shafts inside your oilcloth rucksack.
2021-02-14 17:11:50 -0500:[bolts]>tap my flights
2021-02-14 17:11:51 -0500:You tap some straight bolt flights inside your oilcloth rucksack.
2021-02-14 17:11:51 -0500:[bolts]>inv search glue
2021-02-14 17:11:51 -0500:You rummage about your person, looking for something...
2021-02-14 17:11:51 -0500:  Your deep-orange glue is in a swirling eddy of incandescent light bound by a gold-striated coralite frame.
2021-02-14 17:11:51 -0500:  Your deep-orange glue is in a swirling eddy of incandescent light bound by a gold-striated coralite frame.
2021-02-14 17:11:51 -0500:  Your deep-orange glue is in a swirling eddy of incandescent light bound by a gold-striated coralite frame.
2021-02-14 17:11:51 -0500:  Your wood glue is in an oilcloth rucksack.
2021-02-14 17:11:51 -0500:[Use INVENTORY HELP for more options.]
2021-02-14 17:11:51 -0500:Roundtime: 5 sec.
2021-02-14 17:11:56 -0500:[bolts]>get my glue
2021-02-14 17:11:57 -0500:You get some wood glue from inside your oilcloth rucksack.
2021-02-14 17:11:57 -0500:[bolts]>count my glue
2021-02-14 17:11:57 -0500:The wood glue has 27 uses remaining.
2021-02-14 17:11:57 -0500:[bolts]>put my glue in my rucksack
2021-02-14 17:11:58 -0500:You put your glue in your oilcloth rucksack.
2021-02-14 17:11:58 -0500:[bolts]>tap my shafts
2021-02-14 17:11:58 -0500:You tap some bolt shafts inside your oilcloth rucksack.
2021-02-14 17:11:58 -0500:[bolts]>tap my boltheads
2021-02-14 17:11:58 -0500:You tap some cougar-claw boltheads inside your oilcloth rucksack.
2021-02-14 17:11:59 -0500:--- Lich: tinker active.
2021-02-14 17:11:59 -0500:--- Lich: buff active.
2021-02-14 17:11:59 -0500:--- Lich: buff has exited.
2021-02-14 17:12:01 -0500:[tinker]>get my tinkering book
2021-02-14 17:12:01 -0500:You get a book of master tinkering instructions from inside your oilcloth rucksack.
2021-02-14 17:12:01 -0500:[tinker]>turn my book to chapter 7
2021-02-14 17:12:01 -0500:You turn your book to chapter 7, entitled "Boltmaking".
2021-02-14 17:12:01 -0500:[tinker]>read my book
2021-02-14 17:12:02 -0500:              -=    Chapter 7 : Boltmaking    =-
2021-02-14 17:12:02 -0500: Page 1: bolthead shaping                        Page 9: basilisk bolts
2021-02-14 17:12:02 -0500: Page 2: bolt creation                           Page 10: elsralael bolts
2021-02-14 17:12:02 -0500: Page 3: bolt shaft shaping                      Page 11: soot-stained bolts
2021-02-14 17:12:02 -0500: Page 4: cougar-claw bolts                       Page 12: ice-adder bolts
2021-02-14 17:12:02 -0500: Page 5: boar-tusk bolts                         Page 13: jagged-horn bolts
2021-02-14 17:12:02 -0500: Page 6: sabretooth bolts                        Page 14: drake-fang bolts
2021-02-14 17:12:02 -0500: Page 7: angiswaerd bolts                        
2021-02-14 17:12:02 -0500: Page 8: hele'la bolts                           
2021-02-14 17:12:02 -0500:[tinker]>turn my book to page 4
2021-02-14 17:12:02 -0500:You turn your book to page 4, instructions for crafting cougar-claw bolts.
2021-02-14 17:12:02 -0500:[tinker]>study my book
2021-02-14 17:12:02 -0500:You scan the cougar-claw bolts instructions with a glance and completely understand all facets of the design.
2021-02-14 17:12:02 -0500:You now feel ready to begin the crafting process.
2021-02-14 17:12:02 -0500:Roundtime: 6 sec.
2021-02-14 17:12:08 -0500:[tinker]>put my book in my rucksack
2021-02-14 17:12:08 -0500:You put your book in your oilcloth rucksack.
2021-02-14 17:12:08 -0500:[tinker]>untie my drawknife from my carpenter's belt
2021-02-14 17:12:08 -0500:You untie a sharpened steel drawknife from the straps of your carpenter's belt.
2021-02-14 17:12:09 -0500:[tinker]>get my shafts
2021-02-14 17:12:09 -0500:You get some bolt shafts from inside your oilcloth rucksack.
2021-02-14 17:12:09 -0500:[tinker]>tie my drawknife to my carpenter's belt
2021-02-14 17:12:09 -0500:You attach a sharpened steel drawknife to a strap on your carpenter's belt.
2021-02-14 17:12:09 -0500:[tinker]>untie my shaper from my carpenter's belt
2021-02-14 17:12:09 -0500:You untie a wood shaper from the straps of your carpenter's belt.
2021-02-14 17:12:10 -0500:[tinker]>shape my shafts with my shaper
2021-02-14 17:12:10 -0500:You flip some unfinished cougar-claw bolts over and begin to shape it with your shaper.
2021-02-14 17:12:10 -0500:The bolts are ready for an application of glue to affix each bolthead.
2021-02-14 17:12:10 -0500:Roundtime: 11 sec.
2021-02-14 17:12:10 -0500:You need another cougar-claw boltheads to continue crafting some unfinished cougar-claw bolts.  You believe you can assemble the two ingredients together once you acquire them.
2021-02-14 17:12:10 -0500:[Ingredients can be added by using ASSEMBLE Ingredient1 WITH Ingredient2]
2021-02-14 17:12:21 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:12:21 -0500:You attach a wood shaper to a strap on your carpenter's belt.
2021-02-14 17:12:21 -0500:[tinker]>untie my shaper from my carpenter's belt
2021-02-14 17:12:22 -0500:You untie a wood shaper from the straps of your carpenter's belt.
2021-02-14 17:12:22 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:12:22 -0500:You attach a wood shaper to a strap on your carpenter's belt.
2021-02-14 17:12:22 -0500:[tinker]>get my bolthead
2021-02-14 17:12:22 -0500:You get some cougar-claw boltheads from inside your oilcloth rucksack.
2021-02-14 17:12:22 -0500:[tinker]>assemble my bolts with my bolthead
2021-02-14 17:12:22 -0500:You place your boltheads with your cougar-claw bolts and carefully mark where it will attach when you continue crafting.
2021-02-14 17:12:23 -0500:[tinker]>put my bolthead in my rucksack
2021-02-14 17:12:23 -0500:With a groan, Talia stretches before returning to work.
2021-02-14 17:12:23 -0500:You put your boltheads in your oilcloth rucksack.
2021-02-14 17:12:23 -0500:[tinker]>untie my wood shaper from my carpenter's belt
2021-02-14 17:12:23 -0500:You untie a wood shaper from the straps of your carpenter's belt.
2021-02-14 17:12:23 -0500:[tinker]>shape my bolts with my shaper
2021-02-14 17:12:23 -0500:The shaper doesn't appear suitable for working on some unfinished cougar-claw bolts.
2021-02-14 17:12:24 -0500:[tinker]>analyze my bolts
2021-02-14 17:12:24 -0500:You analyze every minute detail of the cougar-claw bolts and smile knowingly to yourself.
2021-02-14 17:12:24 -0500:When crafting is complete these will be a type of animal part bolts.
2021-02-14 17:12:24 -0500:The bolts must have glue applied to them for affixing the boltheads.
2021-02-14 17:12:24 -0500:From the progress so far, it looks like the bolts are masterfully-crafted (12/12).
2021-02-14 17:12:24 -0500:The bolts are a simple piece to make.
2021-02-14 17:12:24 -0500:You estimate that they are one-fifth done.
2021-02-14 17:12:24 -0500:Roundtime: 10 sec.
2021-02-14 17:12:33 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:12:33 -0500:...wait 1 seconds.
2021-02-14 17:12:33 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:12:33 -0500:...wait 1 seconds.
2021-02-14 17:12:34 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:12:34 -0500:You attach a wood shaper to a strap on your carpenter's belt.
2021-02-14 17:12:34 -0500:[tinker]>get my glue
2021-02-14 17:12:35 -0500:You get some wood glue from inside your oilcloth rucksack.
2021-02-14 17:12:35 -0500:[tinker]>apply my glue to my bolts
2021-02-14 17:12:35 -0500:You apply some glue to your bolts and affix each bolthead in place.
2021-02-14 17:12:35 -0500:Roundtime: 15 sec.
2021-02-14 17:12:49 -0500:[tinker]>put my wood glue in my rucksack
2021-02-14 17:12:49 -0500:...wait 1 seconds.
2021-02-14 17:12:49 -0500:[tinker]>put my wood glue in my rucksack
2021-02-14 17:12:49 -0500:...wait 1 seconds.
2021-02-14 17:12:50 -0500:[tinker]>put my wood glue in my rucksack
2021-02-14 17:12:50 -0500:You put your glue in your oilcloth rucksack.
2021-02-14 17:12:50 -0500:[tinker]>untie my shaper from my carpenter's belt
2021-02-14 17:12:50 -0500:You untie a wood shaper from the straps of your carpenter's belt.
2021-02-14 17:12:50 -0500:[tinker]>shape my bolts with my shaper
2021-02-14 17:12:50 -0500:You flip some unfinished cougar-claw bolts over and begin to shape it with your shaper.
2021-02-14 17:12:50 -0500:Roundtime: 19 sec.
2021-02-14 17:13:00 -0500:A large caravan creaks and shudders as workers unload rock slabs from within.
2021-02-14 17:13:09 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:13:09 -0500:You attach a wood shaper to a strap on your carpenter's belt.
2021-02-14 17:13:09 -0500:[tinker]>untie my shaper from my carpenter's belt
2021-02-14 17:13:09 -0500:You untie a wood shaper from the straps of your carpenter's belt.
2021-02-14 17:13:10 -0500:[tinker]>shape my bolts with my shaper
2021-02-14 17:13:10 -0500:Using abrupt motions you shape a series of grooves into your bolt shafts.
2021-02-14 17:13:10 -0500:Roundtime: 17 sec.
2021-02-14 17:13:14 -0500:Sir Waltharius wandered over from a well-used road.
2021-02-14 17:13:15 -0500:Sir Waltharius runs west.
2021-02-14 17:13:23 -0500:Talia hums softly to herself as she walks away.
2021-02-14 17:13:27 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:13:27 -0500:Sir Waltharius just arrived.
2021-02-14 17:13:27 -0500:You attach a wood shaper to a strap on your carpenter's belt.
2021-02-14 17:13:27 -0500:[tinker]>untie my shaper from my carpenter's belt
2021-02-14 17:13:27 -0500:You untie a wood shaper from the straps of your carpenter's belt.
2021-02-14 17:13:28 -0500:[tinker]>shape my bolts with my shaper
2021-02-14 17:13:28 -0500:You lap the bolt shafts with your shaper and add gentle ribs to the wood surface.
2021-02-14 17:13:28 -0500:A handful of rough edges require carving with a knife to remove.
2021-02-14 17:13:28 -0500:Roundtime: 13 sec.
2021-02-14 17:13:29 -0500:Waltharius puts his lumber in his journey pack.
2021-02-14 17:13:30 -0500:Sir Waltharius wandered over to a well-used road.
2021-02-14 17:13:40 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:13:40 -0500:...wait 1 seconds.
2021-02-14 17:13:40 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:13:40 -0500:...wait 1 seconds.
2021-02-14 17:13:41 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:13:41 -0500:You attach a wood shaper to a strap on your carpenter's belt.
2021-02-14 17:13:41 -0500:[tinker]>untie my shaper from my carpenter's belt
2021-02-14 17:13:41 -0500:You untie a wood shaper from the straps of your carpenter's belt.
2021-02-14 17:13:41 -0500:[tinker]>shape my bolts with my shaper
2021-02-14 17:13:42 -0500:The shaper doesn't appear suitable for working on some unfinished cougar-claw bolts.
2021-02-14 17:13:42 -0500:[tinker]>analyze my bolts
2021-02-14 17:13:42 -0500:You analyze every minute detail of the cougar-claw bolts and smile knowingly to yourself.
2021-02-14 17:13:42 -0500:When crafting is complete these will be a type of animal part bolts.
2021-02-14 17:13:42 -0500:Several adjustments must be made to the bolt by carving it with a knife.
2021-02-14 17:13:42 -0500:From the progress so far, it looks like the bolts are masterfully-crafted (12/12).
2021-02-14 17:13:42 -0500:The bolts are a simple piece to make.
2021-02-14 17:13:42 -0500:You estimate that they are half done.
2021-02-14 17:13:42 -0500:Roundtime: 10 sec.
2021-02-14 17:13:51 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:13:51 -0500:...wait 1 seconds.
2021-02-14 17:13:51 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:13:51 -0500:...wait 1 seconds.
2021-02-14 17:13:52 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:13:52 -0500:You attach a wood shaper to a strap on your carpenter's belt.
2021-02-14 17:13:52 -0500:[tinker]>untie my carving knife from my carpenter's belt
2021-02-14 17:13:52 -0500:You untie a steel carving knife from the straps of your carpenter's belt.
2021-02-14 17:13:53 -0500:[tinker]>carve my bolts with my knife
2021-02-14 17:13:53 -0500:You whittle away the rough edges on your bolts using a steel carving knife.
2021-02-14 17:13:53 -0500:Roundtime: 9 sec.
2021-02-14 17:14:01 -0500:[tinker]>tie my carving knife to my carpenter's belt
2021-02-14 17:14:01 -0500:...wait 1 seconds.
2021-02-14 17:14:01 -0500:[tinker]>tie my carving knife to my carpenter's belt
2021-02-14 17:14:01 -0500:...wait 1 seconds.
2021-02-14 17:14:02 -0500:[tinker]>tie my carving knife to my carpenter's belt
2021-02-14 17:14:02 -0500:You attach a steel carving knife to a strap on your carpenter's belt.
2021-02-14 17:14:02 -0500:[tinker]>untie my shaper from my carpenter's belt
2021-02-14 17:14:02 -0500:You untie a wood shaper from the straps of your carpenter's belt.
2021-02-14 17:14:02 -0500:[tinker]>shape my bolts with my shaper
2021-02-14 17:14:03 -0500:You lap the bolt shafts with your shaper and add gentle ribs to the wood surface.
2021-02-14 17:14:03 -0500:The bolts are ready for an application of glue to attach the flights.
2021-02-14 17:14:03 -0500:Roundtime: 17 sec.
2021-02-14 17:14:03 -0500:You need another bolt flights to continue crafting some unfinished cougar-claw bolts.  You believe you can assemble the two ingredients together once you acquire them.
2021-02-14 17:14:03 -0500:[Ingredients can be added by using ASSEMBLE Ingredient1 WITH Ingredient2]
2021-02-14 17:14:20 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:14:21 -0500:You attach a wood shaper to a strap on your carpenter's belt.
2021-02-14 17:14:21 -0500:[tinker]>untie my shaper from my carpenter's belt
2021-02-14 17:14:21 -0500:You untie a wood shaper from the straps of your carpenter's belt.
2021-02-14 17:14:21 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:14:22 -0500:You attach a wood shaper to a strap on your carpenter's belt.
2021-02-14 17:14:22 -0500:[tinker]>get my flights
2021-02-14 17:14:22 -0500:You get some straight bolt flights from inside your oilcloth rucksack.
2021-02-14 17:14:22 -0500:[tinker]>assemble my bolts with my flights
2021-02-14 17:14:23 -0500:You place your flights with your cougar-claw bolts and carefully mark where it will attach when you continue crafting.
2021-02-14 17:14:23 -0500:[tinker]>untie my wood shaper from my carpenter's belt
2021-02-14 17:14:23 -0500:You untie a wood shaper from the straps of your carpenter's belt.
2021-02-14 17:14:23 -0500:[tinker]>shape my bolts with my shaper
2021-02-14 17:14:23 -0500:The shaper doesn't appear suitable for working on some unfinished cougar-claw bolts.
2021-02-14 17:14:23 -0500:[tinker]>analyze my bolts
2021-02-14 17:14:24 -0500:You analyze every minute detail of the cougar-claw bolts and smile knowingly to yourself.
2021-02-14 17:14:24 -0500:When crafting is complete these will be a type of animal part bolts.
2021-02-14 17:14:24 -0500:The bolts must have glue applied to them for affixing the flights.
2021-02-14 17:14:24 -0500:From the progress so far, it looks like the bolts are masterfully-crafted (12/12).
2021-02-14 17:14:24 -0500:The bolts are a simple piece to make.
2021-02-14 17:14:24 -0500:You estimate that they are close to done.
2021-02-14 17:14:24 -0500:Roundtime: 10 sec.
2021-02-14 17:14:34 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:14:35 -0500:You attach a wood shaper to a strap on your carpenter's belt.
2021-02-14 17:14:35 -0500:[tinker]>get my glue
2021-02-14 17:14:36 -0500:You get some wood glue from inside your oilcloth rucksack.
2021-02-14 17:14:36 -0500:[tinker]>apply my glue to my bolts
2021-02-14 17:14:37 -0500:You dab the surface of your bolts with glue and attach the boltheads to them.
2021-02-14 17:14:37 -0500:Now the flights are ready to be trimmed with a carving knife.
2021-02-14 17:14:37 -0500:Roundtime: 15 sec.
2021-02-14 17:14:51 -0500:[tinker]>put my wood glue in my rucksack
2021-02-14 17:14:51 -0500:...wait 1 seconds.
2021-02-14 17:14:51 -0500:[tinker]>put my wood glue in my rucksack
2021-02-14 17:14:51 -0500:...wait 1 seconds.
2021-02-14 17:14:52 -0500:[tinker]>put my wood glue in my rucksack
2021-02-14 17:14:52 -0500:You put your glue in your oilcloth rucksack.
2021-02-14 17:14:52 -0500:[tinker]>untie my shaper from my carpenter's belt
2021-02-14 17:14:53 -0500:You untie a wood shaper from the straps of your carpenter's belt.
2021-02-14 17:14:53 -0500:[tinker]>shape my bolts with my shaper
2021-02-14 17:14:53 -0500:The shaper doesn't appear suitable for working on some unfinished cougar-claw bolts.
2021-02-14 17:14:53 -0500:[tinker]>analyze my bolts
2021-02-14 17:14:53 -0500:You analyze every minute detail of the cougar-claw bolts and smile knowingly to yourself.
2021-02-14 17:14:53 -0500:When crafting is complete these will be a type of animal part bolts.
2021-02-14 17:14:53 -0500:The arrows must be carved with a knife to trim and shape the flights.
2021-02-14 17:14:53 -0500:From the progress so far, it looks like the bolts are masterfully-crafted (12/12).
2021-02-14 17:14:53 -0500:The bolts are a simple piece to make.
2021-02-14 17:14:53 -0500:You estimate that they are very nearly done.
2021-02-14 17:14:53 -0500:Roundtime: 10 sec.
2021-02-14 17:15:03 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:15:04 -0500:You attach a wood shaper to a strap on your carpenter's belt.
2021-02-14 17:15:04 -0500:[tinker]>untie my carving knife from my carpenter's belt
2021-02-14 17:15:05 -0500:You untie a steel carving knife from the straps of your carpenter's belt.
2021-02-14 17:15:05 -0500:[tinker]>carve my bolts with my knife
2021-02-14 17:15:05 -0500:Using slow strokes you scrape away some rough edges on the bolts using a steel carving knife.
2021-02-14 17:15:05 -0500:Roundtime: 14 sec.
2021-02-14 17:15:05 -0500:Applying the final touches, you complete working on some cougar-claw bolts.
2021-02-14 17:15:18 -0500:[tinker]>tie my carving knife to my carpenter's belt
2021-02-14 17:15:18 -0500:...wait 1 seconds.
2021-02-14 17:15:18 -0500:[tinker]>tie my carving knife to my carpenter's belt
2021-02-14 17:15:18 -0500:...wait 1 seconds.
2021-02-14 17:15:19 -0500:[tinker]>tie my carving knife to my carpenter's belt
2021-02-14 17:15:19 -0500:You attach a steel carving knife to a strap on your carpenter's belt.
2021-02-14 17:15:19 -0500:[tinker]>untie my shaper from my carpenter's belt
2021-02-14 17:15:20 -0500:You untie a wood shaper from the straps of your carpenter's belt.
2021-02-14 17:15:20 -0500:[tinker]>shape my bolts with my shaper
2021-02-14 17:15:20 -0500:You cannot figure out how to do that.  Perhaps finding suitable ingredients and studying some instructions would help.
2021-02-14 17:15:20 -0500:[tinker]>analyze my bolts
2021-02-14 17:15:20 -0500:You analyze every minute detail of the cougar-claw bolts and smile knowingly to yourself.
2021-02-14 17:15:20 -0500:These appear to be a type of animal part bolts that are masterfully-crafted (12/12).
2021-02-14 17:15:20 -0500:The bolts are a simple piece to make.
2021-02-14 17:15:20 -0500:It was made by someone with abilities close to your own skill.
2021-02-14 17:15:20 -0500:You find it impossible to identify who crafted this item.
2021-02-14 17:15:20 -0500:Roundtime: 10 sec.
2021-02-14 17:15:30 -0500:[tinker]>tie my wood shaper to my carpenter's belt
2021-02-14 17:15:30 -0500:You attach a wood shaper to a strap on your carpenter's belt.
2021-02-14 17:15:30 -0500:[tinker]>stow lumber
2021-02-14 17:15:31 -0500:Stow what?  Type 'STOW HELP' for details.
2021-02-14 17:15:31 -0500:[tinker]>get my stamp
2021-02-14 17:15:31 -0500:You get a basic steel stamp from inside your oilcloth rucksack.
2021-02-14 17:15:31 -0500:[tinker]>mark my bolts with my stamp
2021-02-14 17:15:32 -0500:You cannot figure out how to do that.
2021-02-14 17:15:33 -0500:[tinker]>put my stamp in my rucksack
2021-02-14 17:15:33 -0500:You put your stamp in your oilcloth rucksack.
2021-02-14 17:15:33 -0500:[tinker]>put my bolts in my rucksack
2021-02-14 17:15:34 -0500:You put your bolts in your oilcloth rucksack.
2021-02-14 17:15:34 -0500:--- Lich: tinker has exited.
2021-02-14 17:15:34 -0500:[bolts]>stow shafts
2021-02-14 17:15:35 -0500:Stow what?  Type 'STOW HELP' for details.
2021-02-14 17:15:35 -0500:--- Lich: bolts has exited.
```
</details>

### CODE REVIEW

File: bolts.lic [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  def store_tools
+    wait_for_script_to_complete('clerk-tools', ['engineering', 'store'])
+  end
+
+  def order_lumber
+    return if exists?("#{@lumber} lumber") || exists?('shafts')
+    wood = find_piece("#{@lumber} lumber")
+    2.times do
+      order_item(@hometown_data['stock-room'], wood)
+      bput("get #{@lumber} lumber from #{@bag}", "You get", "What were")
```
Comment (KatoakDR) [inline bolts.lic:82]: Per convention in other scripts and to more easily know where functions are source from, may want to prefix these as `DRC.bput`

File: bolts.lic [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      bput("combine my lumber", "You combine", "You must")
+      bput("stow lumber", "You put")
+    end
+  end
+
+  def check_flights
+    return if exists?('flights')
```
Comment (KatoakDR) [inline bolts.lic:89]: Same note about prefixing functions -- this might be in either DRC or DRCI

File: bolts.lic [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    bput("shape lumber into bolt shaft", 'Roundtime')
+    stow_item('shaper')
+    bput("stow my shafts in my #{@bag}", "You put")
+  end
+
+  def make_bolts
+    return if exists?('boltheads')
+    bolt_head = { "cougar-claw" => "curved claw", "boar-tusk" => "boar tusk", "sabertooth" => "ivory fang", "angiswaerd" => "angiswaerd tooth", "hele'la" => "needlesharp tooth", "basilisk" => "basilisk fang", "elsralael" => "elsralael tooth", "soot-stained" => "soot-streaked fang", "ice-adder" => "adder fang", "jagged-horn" => "jagged horn", "drake-fang" => "drake fang" }
+    new_type = bolt_head[@type]
+    case bput("get #{new_type} from my #{@bag}", 'You get', 'I could not', 'What were')
```
Comment (KatoakDR) [inline bolts.lic:125]: Consider `DRCI.get_item(new_type, @bag)`

File: bolts.lic [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    when /You get/
+      get_item('shaper')
+      bput("shape #{new_type} into bolthead", 'Roundtime', 'You fumble')
+      #bput("combine", "You combine") if bput("Get boltheads from my #{@bag}", "You get", "What were") =~ /You get/
+      stow_item('boltheads')
+    when /I could not/, /What were/
+      beep
+      echo("***Out of #{@type}!***")
```
Comment (KatoakDR) [inline bolts.lic:133]: For warnings, consider `DRC.message` instead of `echo`

File: bolts.lic (hunk 3 above)
Comment (Dartellum) [reply bolts.lic:125]: Would need to verify it works in place of the case.

File: bolts.lic (hunk 4 above)
Comment (Dartellum) [reply bolts.lic:133]: Changed.

File: bolts.lic (hunk 1 above)
Comment (Dartellum) [reply bolts.lic:82]: Added DRC.

File: bolts.lic (hunk 2 above)
Comment (Dartellum) [reply bolts.lic:89]: DRCI added to these and search? calls.

File: bolts.lic [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+        dispose('glue')
+        order_item(info['tool-room'], info['glue-number'])
+      end
+    else
+      order_item(info['tool-room'], info['glue-number'])
+    end
+    stow_item('glue')
+  end
+
+  def make_shafts
```
Comment (rpherbig) [inline bolts.lic:112]: Does this also need to stow the lumber?

File: bolts.lic [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+  def make_bolts
+    return if exists?('boltheads')
+    bolt_head = { "cougar-claw" => "curved claw", "boar-tusk" => "boar tusk", "sabertooth" => "ivory fang", "angiswaerd" => "angiswaerd tooth", "hele'la" => "needlesharp tooth", "basilisk" => "basilisk fang", "elsralael" => "elsralael tooth", "soot-stained" => "soot-streaked fang", "ice-adder" => "adder fang", "jagged-horn" => "jagged horn", "drake-fang" => "drake fang" }
+    new_type = bolt_head[@type]
+    case bput("get #{new_type} from my #{@bag}", 'You get', 'I could not', 'What were')
+    when /You get/
+      get_item('shaper')
+      bput("shape #{new_type} into bolthead", 'Roundtime', 'You fumble')
+      #bput("combine", "You combine") if bput("Get boltheads from my #{@bag}", "You get", "What were") =~ /You get/
```
Comment (rpherbig) [inline bolts.lic:129]: Remove commented out code?

File: bolts.lic [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+      wait_for_script_to_complete('tinker', ['continue', 'bolts'])
+      exit_tasks
+      exit
+    end
+
+    @type = args.type
+    @lumber = args.lumber
+    @tools = args.tools
+
+    get_tools if @tools
```
Comment (rpherbig) [inline bolts.lic:54]: It's minor, but I'd rather see a guard clause inside `get_tools` and `store_tools`

File: bolts.lic (hunk 6 above)
Comment (Dartellum) [reply bolts.lic:129]: Removed.

File: bolts.lic (hunk 7 above)
Comment (Dartellum) [reply bolts.lic:54]: What kind of guard?

File: bolts.lic (hunk 5 above)
Comment (Dartellum) [reply bolts.lic:112]: Typical no lumber is left if you let the script purchase. However, If you have your own, maybe. I can check if larger quantities than the one purchase leave lumber. No one has said anything about this from the `arrow` script.

File: bolts.lic (hunk 3 above)
Comment (Dartellum) [reply bolts.lic:125]: Works.

File: bolts.lic (hunk 7 above)
Comment (Dartellum) [reply bolts.lic:54]: I added this check/guard: `wait_for_script_to_complete('clerk-tools', ['engineering', 'get']) unless DRCI.exists?('shaper')`

and this `wait_for_script_to_complete('clerk-tools', ['engineering', 'store']) if DRCI.exists?('shaper')`

**Comment by Dartellum** [comment]
Hold off on merging this as I have numerous updates I am making this morning based on direct feedback from Kiriawen and Sexylumberjack on Discord in the #engineering channel. The changes also apply to `arrows.lic` so I will put in another PR for that today.

**Comment by Dartellum** [comment]
Going to close this one as `shape.lic` got into it. I am going to open a new PR. The scripts are fully tested with others from Discord.

**Pull request closed without merging**